frontend/web/src/features/bookingengine/listingdetails/components/imageGallery.js

```diff
@@ -1,71 +1,102 @@
   const [showOverlay, setShowOverlay] = useState(false);
+  const [activeIndex, setActiveIndex] = useState(0);
 
-  const handleMainImageClick = () => {
-    setShowOverlay(true); // Show the overlay
-  };
-
-  const handleCloseOverlay = () => {
-    setShowOverlay(false); // Hide the overlay
-  };
+  // Prepare main + thumbnails
+  const main = images[0];
+  const thumbs = images.slice(1, 5); // max 4
 
   const toSrc = (key) => {
-    if (!key) return '';
-    if (typeof key === 'string' && (key.startsWith('http://') || key.startsWith('https://'))) {
+    if (!key) return "";
+    if (typeof key === "string" &&
+      (key.startsWith("http://") || key.startsWith("https://"))) {
       return key;
     }
     return `https://accommodation.s3.eu-north-1.amazonaws.com/${key}`;
   };
 
+  const nextImage = () => {
+    setActiveIndex((prev) => (prev + 1) % images.length);
+  };
+
+  const prevImage = () => {
+    setActiveIndex((prev) =>
+      prev === 0 ? images.length - 1 : prev - 1
+    );
+  };
+
   return (
     <section className="image-section">
+
+      {/* GALLERY */}
       <div className="image-gallery">
-        <img
-          className="main-image"
-          src={toSrc(images[0].key)}
-          alt="Main"
-          onClick={handleMainImageClick} // Show overlay on click
-        />
+
+        {/* MAIN IMAGE */}
+        {main && (
+          <img
+            className="main-image"
+            src={toSrc(main.key)}
+            onClick={() => { setActiveIndex(0); setShowOverlay(true); }}
+          />
+        )}
+
+        {/* THUMBNAILS */}
         <div className="small-images-container">
-          {images.map((image) => {
-            if (image !== images[0]) {
-              return (
-                <img
-                  key={image.key}
-                  className="small-image"
-                  src={toSrc(image.key)}
-                  alt={`Extra ${images.indexOf(image)}`}
-                />
-              );
-            }
-          })}
+          {thumbs.map((img, index) => (
+            <img
+              key={img.key}
+              className="small-image"
+              src={toSrc(img.key)}
+              onClick={() => { setActiveIndex(index + 1); setShowOverlay(true); }}
+            />
+          ))}
         </div>
+
       </div>
-      {/* Rating and host label removed per request */}
 
-      {/* Overlay */}
+      {/* FULLSCREEN OVERLAY */}
       {showOverlay && (
         <div className="image-overlay">
-          <button className="close-overlay-button" onClick={handleCloseOverlay}>
-            Close
+
+          <button className="close-overlay-button" onClick={() => setShowOverlay(false)}>
+            ×
           </button>
-          <div className="overlay-images">
-            {images.map((image, index) => (
+
+          <div className="overlay-center-wrapper">
+
+            <button className="nav-button left" onClick={prevImage}>
+              ‹
+            </button>
```

Comment:
Wrapping these lines to their own lines doesn't increase readability since it's only one character. You might want to consider putting this on one line.
```suggestion
            <button className="nav-button left" onClick={prevImage}> ‹ </button>
```